File: templates/csharp/mavlink_dialect.cs
namespace Mavlink;

public abstract class MavlinkDialect
{
    public abstract int Version { get; }

    public abstract MavlinkMessage? Parse(int messageId, ReadOnlySpan<byte> data);

    /// <summary>Return CRC extra for messageId, or -1 if unsupported.</summary>
    public abstract int CrcExtra(int messageId);
}

File: templates/py/mavlink_dialect.py
from __future__ import annotations

from abc import ABC, abstractmethod

from mavlink_message import MavlinkMessage


class MavlinkDialect(ABC):
    @property
    @abstractmethod
    def version(self) -> int:
        ...

    @abstractmethod
    def parse(self, message_id: int, data: bytes) -> MavlinkMessage | None:
        ...

    @abstractmethod
    def crc_extra(self, message_id: int) -> int:
        """Return CRC extra for message_id, or -1 if unsupported."""
